### PR TITLE
Add support for highlighted path

### PR DESCRIPTION
**Pull request by jonathan-marmelab**
This closes the following user story : https://trello.com/c/nxa4cjQ6/27-5-as-robert-given-i-have-to-move-my-pawn-i-want-to-only-see-the-locations-i-can-reach

depends on #8

### CODE REVIEW

File: internal/model/board.go [hunk 1]
```diff
@@ -259,38 +293,143 @@ func (b *Board) MoveCurrentPlayerTo(line, row int) error {
 	return nil
 }
 
-// CurrentPlayer returns the current player.
-func (b Board) CurrentPlayer() *Player {
+func (b Board) GetCurrentPlayer() *Player {
```
Comment (arimet) [inline internal/model/board.go:296]: Some function start in Uppercase, other not. is there an explanation ?

File: internal/model/board.go [hunk 2]
```diff
@@ -259,38 +293,143 @@ func (b *Board) MoveCurrentPlayerTo(line, row int) error {
...
+		currentPlayer = b.GetCurrentPlayer()
+
+		size  = b.GetSize()
+		graph = dijkstra.NewGraph()
+
+		// targetNode is the target node, defaults to remaining tile ID.
+		targetNode = size*size + 1
+
+		// reverseCoordinates holds the coordinates for a tile ID.
+		reverseCoordinates = make(map[int]*Coordinate)
```
Comment (arimet) [inline internal/model/board.go:391]: I don't understand the meaning of this naming

File: internal/model/player.go [hunk 3]
```diff
@@ -15,11 +15,8 @@ type Player struct {
 	// Color is the player color
 	Color Color `json:"color"`
 
-	// Line is the player current line.
-	Line int `json:"line"`
-
-	// Row is the current player row.
-	Row int `json:"row"`
+	// position is the player current position
+	Position *Coordinate `json:"position"`
```
Comment (arimet) [inline internal/model/player.go:19]: :ok_hand:

File: internal/model/board.go (hunk 1 above)
Comment (jonathan-marmelab) [reply internal/model/board.go:296]: This is du to the Go language design : Upper-Cased functyionas are public, Lower-Cased are private.

File: internal/model/tile_exit.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		TileExitBottom: TileExitTop,
+		TileExitLeft:   TileExitRight,
+	}
+)
+
+func (t TileExit) Opposite() TileExit {
+	return oppositeTileExit[t]
+}
+
+func (t TileExit) ExitCoordinate(line, row int) *Coordinate {
```
Comment (arimet) [inline internal/model/tile_exit.go:26]: This could also be a data structure

File: internal/model/board.go (hunk 2 above)
Comment (jonathan-marmelab) [reply internal/model/board.go]: Done, I added a function to build the graph and renamed this to `getCoordinatesByVertex`

**Review by arimet — APPROVED** [review]

**Pull request merged**